tests/system/src/test/java/test/com/sun/marlin/ScaleX0Test.java

```diff
@@ -0,0 +1,226 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.com.sun.marlin;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.lang.reflect.Method;
+
+import java.util.Locale;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.beans.value.ObservableValue;
+import javafx.scene.Group;
+import javafx.scene.Node;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import javafx.stage.WindowEvent;
+import javafx.scene.control.Button;
+import javafx.scene.control.Slider;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.scene.shape.Rectangle;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import test.util.Util;
+import static test.util.Util.TIMEOUT;
+
+/**
+ * @test
+ * @bug 8312603
+ * @summary Check the crash with MarlinFX renderer if scaleX or scaleY is pure 0.0
+ */
+public class ScaleX0Test {
+
+    private final static int SIZE = 800;
+
+    static CountDownLatch launchLatch = new CountDownLatch(1);
+
+    static final ByteArrayOutputStream out = new ByteArrayOutputStream(1024 * 1024);
+    static PrintStream defaultErrorStream = System.err;
+
+    static {
+        Locale.setDefault(Locale.US);
+
+        System.setProperty("prism.verbose", "false");
+        // Enable Marlin logging
+        System.setProperty("prism.marlin.log", "true");
+    }
+
+    @BeforeClass
+    public static void setupOnce() throws Exception {
+        defaultErrorStream = System.err;
+        // Capture stderr:
+        System.setErr(new PrintStream(out, true));
+
+        CountDownLatch startupLatch = new CountDownLatch(1);
+        Util.startup(startupLatch, () -> {
+            Platform.setImplicitExit(false);
+            startupLatch.countDown();
+        });
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        System.setErr(defaultErrorStream);
+        Util.shutdown();
+    }
+
+    @Test(timeout = 15000)
+    public void testMarlinAIOOBEwhenScaleXIs0() {
+        Scene scene = createScene();
+
+        Platform.runLater(() -> {
+            Stage stage = new Stage();
+            stage.setScene(scene);
+            stage.addEventHandler(WindowEvent.WINDOW_SHOWN, e -> Platform.runLater(launchLatch::countDown));
```

Comment:
Not mandatory, but can be changed to  `stage.setOnShown(e -> Platform.runLater(launchLatch::countDown));` 
The import  `import javafx.stage.WindowEvent;`  can be removed with this change.